lib/vsopdf/vso_appointment_form.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'pdf_forms'
+require 'tempfile'
+require 'securerandom'
+
+class VsoAppointmentForm
+  def initialize(appt)
+    @appt = appt
+  end
+
+  def to_pdf_args
+    {
+      "nameofvet": @appt.veteranFullName,
+      "SSNno": @appt.veteranSSN,
+      "filenumber": @appt.vaFileNumber,
+      "insno": @appt.insuranceNumber,
+      "claimantname": @appt.claimantFullName,
+      "address": @appt.claimantAddress,
+      "emailaddress": @appt.claimantEmail,
+      "daytime": @appt.claimantDaytimePhone,
+      "eveningphonenumber": @appt.claimantEveningPhone,
+      "relationship": @appt.relationship,
+      "Date@appt": @appt.appointmentDate,
+      "nameofservice": @appt.organizationName,
+      "e-mailaddressoftheorganizationnamedinitem3a": @appt.organizationEmail,
+      "jobtitile": "#{@appt.organizationRepresentativeName}, #{@appt.organizationRepresentativeTitle}",
+      "drugabuse": @appt.disclosureExceptionDrugAbuse ? 1 : 0,
+      "alcoholismoralcoholabuse": @appt.disclosureExceptionAlcoholism ? 1 : 0,
+      "infectionwiththehumanimmunodeficiencyvirushiv": @appt.disclosureExceptionHIV ? 1 : 0,
+      "sicklecellanemia": @appt.disclosureExceptionSickleCellAnemia ? 1 : 0
+    }.map { |k, v| ["F[0].Page_1[0].#{k}[0]", v] }.to_h
+  end
+
+  def generate_pdf
+    tmpf = Tempfile.new(['vsopdf', '.pdf'])
+    PdfForms.new(Settings.binaries.pdftk).fill_form 'lib/vsopdf/VBA-21-22-ARE.pdf', tmpf.path, to_pdf_args
+    tmpf.close
+    tmpf.path
+  end
+
+  def get_metadata(path)
+    {
+      "numberAttachments": 0,
+      "veteranFirstName": @appt.veteranFullName.split(' ').shift,
```

Comment:
Can we reliably get first and last names by splitting the full name into an array? Could the form have first and last name fields that form a full name rather than the other way around?

Reply:
nope, that's a hack until I update the schema